### DISCUSSION TITLE
What is the target state in the merge command for? Can I ignore it for files?

### DISCUSSION THREAD

**Discussion by halloleo** [twpayne/chezmoi#3802] [Q&A] [answered]
Sorry to ask, but I don't really understand what the _target state_ is for in the `merge`command.

Say, I  have the file `~/.config/starship.toml` in chezmoi on machine A

Now I changed the file on another machine B and added the changes to chezmoi's source dir on machine B. Then I synced the source dir back to machine A. 

Now on A I don't want to just apply the changes, but pick and choose "the best of both", i.e. _merge_. On machine A I now have just two different files which I want to merge., the destination file `~/.config/starship.toml` and source dir file `<SOURCEDIR>/dot_config/starship.toml`. 

But vimdiff opens three files, so it includes the target state, but there is, I think no difference between target and one of teh other files Can I just ignore the target state?

After reading about the [Concepts](https://www.chezmoi.io/reference/concepts/) I can only explain this with the idea that the third file is solely needed for templates. So in my case I can just ignore it?

I just save the best of both to `~/.config/starship.toml` , then add to chezmoi, sync and apply on machine B to get everything settled, right?

**Comment by halostatue** [comment]
The target state represents what chezmoi will write given the current source. For non-template files, it should be the same as the source.

**Comment by halloleo** [comment]
I see! Thanks for this.

So for non-template files, the three-way merge here is *not* showing a "base" version (like the git three-way merge).

So for non-template files, I just look at source & destination, save my result to the destination and then go from there, right?

**Reply by twpayne** [reply, marked as answer]
> So for non-template files, the three-way merge here is _not_ showing a "base" version (like the git three-way merge).

Correct. chezmoi is not git.



> So for non-template files, I just look at source & destination, save my result to the destination and then go from there, right?

No. Save the result to the source so that your changes to the destination are included the next time you run `chezmoi apply`.
